### PR TITLE
Api 38534 documentation for transaction

### PR DESCRIPTION
**Pull request by rockwellwindsor-va**
## Summary
* Adds drop downs in sync docs for transaction ID request/responses
* Refactors one reused method's position in the file so it can just to be shared
* _Refactors the meta property to be inserted and handled via the blueprinter in place of the controller lines from the previous commit for the code/test/error handling._

## Related issue(s)
[API-38534](https://jira.devops.va.gov/browse/API-38534)

## Testing done
- [x] *New code is covered by unit tests*

## Screenshots
* REQUEST

Screenshot 2024-07-26 at 10 05 02 AM

* RESPONSE (after blueprinter update `meta` now shows at the bottom of the response)
Screenshot 2024-07-29 at 4 04 20 PM

## What areas of the site does it impact?
	modified:   modules/claims_api/app/swagger/claims_api/v2/dev/swagger.json
	modified:   modules/claims_api/app/swagger/claims_api/v2/production/swagger.json
	modified:   modules/claims_api/spec/requests/v2/veterans/rswag_disability_compensation_request_spec.rb

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [x]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Comment by stiehlrod** [comment]
@rockwellwindsor-va I am not seeing the request body addition of the transaction_id. I do see it on the response
Screenshot 2024-07-26 at 9 36 33 AM

File: modules/claims_api/app/controllers/claims_api/v2/veterans/disability_compensation_controller.rb [hunk 1]
```diff
@@ -110,15 +110,25 @@ def synchronous
...
-          ), status: :accepted, location: url_for(controller: 'claims', action: 'show', id: auto_claim.id)
+          form_blueprint = JSON.parse(ClaimsApi::V2::Blueprints::AutoEstablishedClaimBlueprint.render(
+                                        auto_claim, root: :data, async: false
+                                      ))
+
+          form_meta = { transactionId: auto_claim.transaction_id }
+
+          render_hash = {}
+          render_hash[:meta] = form_meta if auto_claim.transaction_id.present?
+          render_hash[:data] = form_blueprint['data']
```
Comment (FonzMP) [inline modules/claims_api/app/controllers/claims_api/v2/veterans/disability_compensation_controller.rb:121]: I'd prefer this to live in the Blueprinter.

**Comment by rockwellwindsor-va** [comment]
> @rockwellwindsor-va I am not seeing the request body addition of the transaction_id. I do see it on the response Screenshot 2024-07-26 at 9 36 33 AM

Ya I'm not seeing the dropdown in the response image, not sure why.  Also not sure which is selected in the request but I am seeing it in the 'transaction ID' dropdown on my local (which is up to date with the branch)
Screenshot 2024-07-29 at 10 36 04 AM

File: modules/claims_api/app/controllers/claims_api/v2/veterans/disability_compensation_controller.rb (hunk 1 above)
Comment (rockwellwindsor-va) [reply modules/claims_api/app/controllers/claims_api/v2/veterans/disability_compensation_controller.rb:121]: @FonzMP  Ya I did try that.  The issue I ran into was the [BluePrinter](https://github.com/procore-oss/blueprinter) wants a root object sent in, but there is no root for this.  `Meta` and `Data` are both at the same level.  Perhaps we can look at this and see if I am missing something.

I do think the one thing I got to work using the BluePrinter as having `data` as a root, and then in the controller just doing a return and setting something along the lines of `variable[:data]`  on the return to remove the outer data wrapper, but that felt hacky.

File: modules/claims_api/app/controllers/claims_api/v2/blueprints/meta_blueprint.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+module ClaimsApi
+  module V2
+    module Blueprints
+      class MetaBlueprint < Blueprinter::Base
+        association :data, blueprint: AutoEstablishedClaimBlueprint do |claim, _options|
+          claim
+        end
+
+        field :meta, if: ->(_field_name, claim, _options) { !claim.transaction_id.nil? } do |claim, _options|
```
Comment (FonzMP) [inline modules/claims_api/app/controllers/claims_api/v2/blueprints/meta_blueprint.rb:11]: `!` of `nil?` ?

Comment (rockwellwindsor-va) [reply modules/claims_api/app/controllers/claims_api/v2/blueprints/meta_blueprint.rb:11]: @FonzMP I'm not sure I did that, I need to test but I think Rubocop made that change.  Not totally sure of that but I will need to test tomorrow.

**Review by stiehlrod — DISMISSED** [review]
Tested the async method with and without a transaction_id in the request body.

**Review by stiehlrod — APPROVED** [review]

**Pull request merged**